python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -156,1253 +120,308 @@ class MakeBrighterFatterKernelTaskConfig(pexConfig.Config):
-                    ptcTask._plotPtc(ptcDataset, ptcConfig.ptcFitType, pdfPages)
-
-        # having calculated and applied the gains for all code-paths we can now
-        # generate the kernel(s)
-        self.log.info('Generating kernel(s) for %s' % detNum)
-        for det_object in xcorrs.keys():  # looping over either detectors or amps
-            if self.config.level == 'DETECTOR':
-                objId = 'detector %s' % det_object
-            elif self.config.level == 'AMP':
-                objId = 'detector %s AMP %s' % (detNum, det_object)
-
-            try:
-                meanXcorr, kernel = self.generateKernel(xcorrs[det_object], means[det_object], objId)
-                kernels[det_object] = kernel
-                meanXcorrs[det_object] = meanXcorr
-            except RuntimeError:
-                # bad amps will cause failures here which we want to ignore
-                self.log.warn('RuntimeError during kernel generation for %s' % objId)
-                continue
+        Returns
+        -------
+        results : `lsst.pipe.base.Struct`
+            The resulst struct containing:
 
-        bfKernel = BrighterFatterKernel(self.config.level)
-        bfKernel.means = means
-        bfKernel.rawMeans = rawMeans
-        bfKernel.rawXcorrs = rawXcorrs
-        bfKernel.xCorrs = xcorrs
-        bfKernel.meanXcorrs = meanXcorrs
-        bfKernel.originalLevel = self.config.level
-        try:
-            bfKernel.gain = ptcDataset.gain
-            bfKernel.gainErr = ptcDataset.gainErr
-            bfKernel.noise = ptcDataset.noise
-            bfKernel.noiseErr = ptcDataset.noiseErr
-        except NameError:  # we don't have a ptcDataset to store results from
-            pass
-
-        if self.config.level == 'AMP':
-            bfKernel.ampwiseKernels = kernels
-            ex = self.config.ignoreAmpsForAveraging
-            bfKernel.detectorKernel = bfKernel.makeDetectorKernelFromAmpwiseKernels(detNum, ampsToExclude=ex)
-
-        elif self.config.level == 'DETECTOR':
-            bfKernel.detectorKernel = kernels
-        else:
-            raise RuntimeError('Invalid level for kernel calculation; this should not be possible.')
+            ``outputBfk`` : `lsst.ip.isr.BrighterFatterKernel`
+                Resulting Brighter-Fatter Kernel.
+        """
+        if len(dummy) == 0:
+            self.log.warn("No dummy exposure found.")
 
-        dataRef.put(bfKernel)
+        detector = camera[inputDims['detector']]
+        detName = detector.getName()
 
-        self.log.info('Finished generating kernel(s) for %s' % detNum)
-        return pipeBase.Struct(exitStatus=0)
+        if self.config.level == 'DETECTOR':
+            detectorCorrList = list()
+
+        bfk = BrighterFatterKernel(camera=camera, detectorId=detector.getId(), level=self.config.level)
+        bfk.means = inputPtc.finalMeans  # ADU
+        bfk.variances = inputPtc.finalVars  # ADU^2
+        bfk.rawXcorrs = inputPtc.covariances  # ADU^2
```

Comment:
PTC calls it "covariances" and the old BF code calls them "correlations", and I guess that you figured out that in practice these objects are the same. However, should we try to unify the terms? There's a difference between "covariance" and "correlation", and this might be confusing.

Reply:
Or at least make a comment here.

Reply:
I've added a comment.  I believe the scaling done in the loop changes the covariance to a correlation, and was done in the previous code as part of the PTC-like-step.